### PR TITLE
fix(deps): upgrade react-spring to v9.1.2

### LINKED ISSUE #1290: Line Chart Breaking with NextJS after upgrading to 0.65.1
**Describe/explain the bug**
The lines on my ResponsiveLine component are messed up after upgrading to version 0.65.1. Just to make sure, I downgraded to 0.64.0 and the ResponsiveLine components looked great again.

Console Error with static site hosted locally or on Vercel
```
TypeError: r.willAdvance is not a function
```

**To Reproduce**
The easiest way is to clone the repo below and try running `yarn prod` with each version of `@nivo/line` and you'll see the problems with `0.65.1` locally. I also have this hosted on Vercel (with version `0.65.1`).

[Broken version hosted on Vercel](https://next-nivo.vercel.app/)
[Github Repo](https://github.com/p45mark/next-nivo)
image

Thanks!

### LINKED ISSUE #1453: Charts based on react-spring@9.0.0-rc.3 have prod issue
**Describe/explain the bug**
All charts that are using react-spring@9.0.0-rc.3 throws error on the production [Uncaught TypeError: r.willAdvance is not a function](https://github.com/pmndrs/react-spring/issues/1078)

**To Reproduce**
@nivo/line@0.67.0, 
@nivo/core@0.67.0

1. Reload the page with the chart
2. Open dev tools
4. See an error

**Desktop:**
 - Browser [e.g. chrome]

**Additional context**
I hope this information will help you to resolve the issue. Thank you for your time.

### PR DESCRIPTION
**Pull request by jamesdh**
Upgrades from react-spring @ 9.0.0-rc3 to 9.0.1 

I have to imagine a lot of people stumbled upon this bug and rather than looking for a solution as described in #1290, they simply downgraded. After accidentally running into it again today I decided it's time to see if a fix is available, so here you go!

Fixes #1290 
Fixes #1453

### CODE REVIEW

**Comment by rap2hpoutre** [comment]
Thank you so much @jamesdh! Hopefully, it will be merged soon (not sure the project is still actively maintained though).

**Comment by rogemus** [comment]
@plouc Any chance for you to take a quick look?

**Review by wyze — CHANGES_REQUESTED** [review]
Hi @jamesdh, would you mind upgrading to the latest version of the package? Looks like it is now `9.1.1`. Thanks!

File: packages/annotations/package.json [hunk 1]
```diff
@@ -29,7 +29,7 @@
   "dependencies": {
     "@nivo/colors": "0.67.0",
     "lodash": "^4.17.11",
-    "react-spring": "9.0.0-rc.3"
+    "react-spring": "9.0.1"
```
Comment (rogemus) [inline packages/annotations/package.json:32]: Can we use `^9.1.0` version here and in all other packages as well?

**Comment by wyze** [comment]
@jamesdh I saw you upgraded to the latest and that broke the build. I was working on fixing the errors and didn't know if you were as well. I'll continue working on it and push my changes here.

**Comment by jamesdh** [comment]
@wyze @plouc @rogemus I originally used 9.0.1 because I knew it should just be a drop-in replacement that works for now. 9.1.x was out when I first submitted this PR, but I saw in its release notes that it contained breaking changes and I didn't want to bother with them for the time being. 

I just tried upgrading to 9.1.2 and sure enough, it causes numerous breakages during the build. I say we just merge this for now since [using 9.0.0.rc3 is problematic beyond just this script issue](https://github.com/plouc/nivo/issues/1472#issuecomment-816909820):

> My company has a proxy around NPM that only releases stable packages. They won't release react-spring 9.0.0-rc.3. So I haven't been able to work with the latest versions of Nivo

Then we can then do another PR that contains the extra work to get 9.1.x working.

**Comment by jamesdh** [comment]
> @jamesdh I saw you upgraded to the latest and that broke the build. I was working on fixing the errors and didn't know if you were as well. I'll continue working on it and push my changes here.

@wyze I just force-pushed a revert back to 9.0.1. See my comment above.

**Comment by wyze** [comment]
Thats fine. I reapplied your patch of the upgrade and then did another commit that fixes the type errors. This allowed the build to pass locally, so we will see what CI says.

**Review by wyze — APPROVED** [review]
Looks good! Thanks.

**Pull request merged**